### PR TITLE
Wrap URL string in quotes to avoid escaping issues

### LINKED ISSUE #31224: [Bug]: The preview of the personal wallpapers of the dashboard is not displayed
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When I want to choose a background other than those proposed by default in my dashboard, I click on the button allowing me to choose among the images available in my files. When the popup that allows me to browse and select the file is displayed, I notice that no thumbnails are displayed (despite the fact that the files and folders are correctly listed and accessible). 

**Important : The problem occurs only when the name of the folder containing the wallpapers includes an apostrophe.**
**Check additional information that have been added below recently (with screenshots ...).**

### Steps to reproduce

1. Click on the "Customize" button
2. Pick on the "Pick from files" button
3. The popup appears ; folders and files appear but no image thumbnails appear at all

### Expected behavior

Each image should have a thumbnail.

### Installation method

Web installer

### Operating system

Ubuntu 20.04

### PHP engine version

8.0.15

### Web server

Apache

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

After minor update

### Are you using the Nextcloud Server Encryption module?

No

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nextcloud.mydomain.fr"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "22.2.5.1",
        "overwrite.cli.url": "https:\/\/nextcloud.mydomain.fr",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "default_language": "fr",
        "default_locale": "fr_FR",
        "default_phone_region": "ISO3166-2",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "maintenance": false,
        "theme": "",
        "loglevel": 2,
        "updater.release.channel": "stable",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - accessibility: 1.8.0
  - activity: 2.15.0
  - announcementcenter: 6.1.1
  - bruteforcesettings: 2.3.0
  - calendar: 3.0.6
  - camerarawpreviews: 0.7.15
  - circles: 22.2.0
  - cloud_federation_api: 1.5.0
  - comments: 1.12.0
  - contacts: 4.0.7
  - contactsinteraction: 1.3.0
  - dashboard: 7.2.0
  - dav: 1.20.0
  - extract: 1.3.3
  - federatedfilesharing: 1.12.0
  - federation: 1.12.0
  - files: 1.17.0
  - files_automatedtagging: 1.12.0
  - files_external: 1.13.1
  - files_pdfviewer: 2.3.1
  - files_rightclick: 1.1.0
  - files_sharing: 1.14.0
  - files_trashbin: 1.12.0
  - files_versions: 1.15.0
  - files_videoplayer: 1.11.0
  - files_zip: 1.0.1
  - firstrunwizard: 2.11.0
  - groupfolders: 10.0.2
  - guests: 2.1.0
  - impersonate: 1.9.0
  - logreader: 2.7.0
  - lookup_server_connector: 1.10.0
  - metadata: 0.15.0
  - nextcloud_announcements: 1.11.0
  - notes: 4.3.0
  - notifications: 2.10.1
  - oauth2: 1.10.0
  - password_policy: 1.12.0
  - passwords: 2021.12.20
  - phonetrack: 0.7.0
  - photos: 1.4.0
  - privacy: 1.6.0
  - provisioning_api: 1.12.0
  - quota_warning: 1.13.0
  - recommendations: 1.1.0
  - richdocuments: 4.2.4
  - richdocumentscode: 21.11.103
  - serverinfo: 1.12.0
  - settings: 1.4.0
  - sharebymail: 1.12.0
  - spreed: 12.2.3
  - support: 1.5.0
  - survey_client: 1.10.0
  - systemtags: 1.12.0
  - text: 3.3.0
  - theming: 1.13.0
  - twofactor_backupcodes: 1.11.0
  - twofactor_totp: 6.2.0
  - updatenotification: 1.12.0
  - user_status: 1.2.0
  - viewer: 1.6.0
  - weather_status: 1.2.0
  - workflowengine: 2.4.0
Disabled:
  - admin_audit
  - duplicatefinder: 0.0.13
  - encryption
  - user_ldap
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
Too long - ask me if necessary
```

### Additional info

I think I have always seen this bug as long as the dashboard has existed (since NextCloud 20).

### PR DESCRIPTION
**Pull request by niloc132**
This ensures that the browser will expect that single-quotes in the url string are not the end of the string, but part of the url.

Fixes #31224
Signed-off-by: Colin Alworth <colin@colinalworth.com>

* Resolves: #31224

## Summary
The "file picker" dialog was previously not able to display previews for files if there was a `'` character in their path. This fix wraps the string passed to the css `url()` function with double-quotes, since URLs must escape double quotes (so additional quoting won't be required, whereas if the wrapping was done with a single-quote, then additional single-quotes in the URL would need to be further escaped). This is the simplest fix to ensure that properly-encoded URLs will be recognized by the browser.

## TODO

N/A

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [X] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included (**Appears to not apply, I am not seeing testing for this kind of change, please direct me if this is incorrect**)
- [X] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required (**not required**)
- [X] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes) (**N/A**)

Before:
image

After:
image

There are some other notes that js should be built and checked in, but it isn't clear to me if CI should do that. The current settings for Github CodeSpaces cannot run the `make dev-setup` command in its present configuration, and I haven't figured out the precise combination that should build without changes at current HEAD, so any change I make appears to modify every file in dist/.

### CODE REVIEW

File: core/src/OC/dialogs.js [hunk 1]
```diff
@@ -1272,7 +1272,7 @@ const Dialogs = {
 				var previewUrl = OC.generateUrl('/core/preview.png?') + $.param(urlSpec)
 				img.onload = function() {
 					if (img.width > 5) {
-						$row.find('td.filename').attr('style', 'background-image:url(' + previewUrl + ')')
+						$row.find('td.filename').attr('style', 'background-image:url(\"' + previewUrl + '\")')
```
Comment (Pytal) [inline core/src/OC/dialogs.js:1275]:
```suggestion
						$row.find('td.filename').attr('style', `background-image: url("${previewUrl}")`)
```

**Comment by niloc132** [comment]
Should I also modify the filepicker.html file to include these quotes, even though that part of the template is replaced by this JS?

**Comment by Pytal** [comment]
> Should I also modify the filepicker.html file to include these quotes, even though that part of the template is replaced by this JS?

Sure :)

**Comment by niloc132** [comment]
Just my luck to have a major outage in GHA when I try to make my first PR to this project..

image

https://www.githubstatus.com/

**Review by skjnldsv — APPROVED** [review]

**Review by Pytal — APPROVED** [review]

**Comment by niloc132** [comment]
Anything else I can/should do to help this along?

**Comment by szaimen** [comment]
Yes, please rebase on top of master, run npm run build and commit all files here. Thanks!

**Comment by niloc132** [comment]
I'm afraid I can't follow the setup instructions locally (I'm not going to be giving `sudo` rights to npm), and the codespaces setup doesn't work (wrong npm/node versions, and an hour of trial and error couldn't get it to have the right ones). The closest I could get with codespaces (as noted in my description) would modify every file in `dist/`, so it seems the expected setup is very specific. Is there an accurate setup doc that I can follow, or is there a CI step that can provide the expected output for those files?

**Comment by szaimen** [comment]
Yes, there is a command available that would do this automatically. Unfortunately it does not work for forks.

However did you already try npm ci after the rebase and the npm run build? I unfortunately forgot to mention this.

**Comment by niloc132** [comment]
For what its worth, I created the codespace on the main repo, and only created a fork after I wasn't able to build (and manually edited the already-minified js file to test my fix directly). 

I'll rebase and npm ci, run build and post any updates.

**Comment by szaimen** [comment]
> I'll rebase and npm ci, run build and post any updates.

Hi @niloc132 any update on this? :)

**Comment by skjnldsv** [comment]
Closing as 27 and 28 got a new picker

**Pull request closed without merging**